app/components/post_textbox/post_textbox_base.js

```diff
@@ -408,62 +430,146 @@ export default class PostTextBoxBase extends PureComponent {
                 }],
             );
         } else {
+            const currentMembersCount = this.props.currentChannelMembersCount;
+            const notificationsToChannel = this.props.enableConfirmNotificationsToChannel;
+            const toAllorChannel = this.textContainsAtAllAtChannel(value);
+
             if (value.indexOf('/') === 0) {
                 this.sendCommand(value);
+            } else if (notificationsToChannel && currentMembersCount > NOTIFY_ALL_MEMBERS && toAllorChannel) {
+                this.showSendToAllOrChannelAlert(currentMembersCount);
             } else {
-                const postFiles = files.filter((f) => !f.failed);
-                const post = {
-                    user_id: currentUserId,
-                    channel_id: channelId,
-                    root_id: rootId,
-                    parent_id: rootId,
-                    message: value,
-                };
-
-                actions.createPost(post, postFiles);
-
-                if (postFiles.length) {
-                    actions.handleClearFiles(channelId, rootId);
-                }
+                this.doSubmitMessage();
             }
+        }
+    };
 
-            if (Platform.OS === 'ios') {
-                // On iOS, if the PostTextbox height increases from its
-                // initial height (due to a multiline post or a post whose
-                // message wraps, for example), then when the text is cleared
-                // the PostTextbox height decrease will be animated. This
-                // animation in conjunction with the PostList animation as it
-                // receives the newly created post is causing issues in the iOS
-                // PostList component as it fails to properly react to its content
-                // size changes. While a proper fix is determined for the PostList
-                // component, a small delay in triggering the height decrease
-                // animation gives the PostList enough time to first handle content
-                // size changes from the new post.
-                setTimeout(() => {
-                    this.handleTextChange('');
-                }, 250);
-            } else {
-                this.handleTextChange('');
-            }
+    textContainsAtAllAtChannel = (text) => {
+        const textWithoutCode = text.replace(/(`+)([^`]|[^`][\s\S]*?[^`])\1(?!`)| *(`{3,}|~{3,})[ .]*(\S+)? *\n([\s\S]*?\s*)\3 *(?:\n+|$)/g, '');
+        return (/\B@(all|channel)\b/i).test(textWithoutCode);
+    }
 
-            this.changeDraft('');
+    showSendToAllOrChannelAlert = (currentMembersCount) => {
+        const {intl} = this.context;
+        const {channelTimezoneCount} = this.state;
+        const {isTimezoneEnabled} = this.props;
+        const notifyAllTitle = (
```

Comment:
This constant and the following two can be defined in the constructor as they never change